nixos/modules/services/networking/tox-node.nix

```diff
@@ -0,0 +1,152 @@
+    { pk = "D527E5847F8330D628DAB1814F0A422F6DC9D0A300E6C357634EE2DA88C35463";
+      addr = "tox.novg.net:33445";
+    }
+    { pk = "257744DBF57BE3E117FE05D145B5F806089428D4DCE4E3D0D50616AA16D9417E";
+      addr = "95.31.18.227:33445";
+    }
+    { pk = "2555763C8C460495B14157D234DD56B86300A2395554BCAE4621AC345B8C1B1B";
+      addr = "185.14.30.213:443";
+    }
+    { pk = "BEF0CFB37AF874BD17B9A8F9FE64C75521DB95A37D33C5BDB00E9CF58659C04F";
+      addr = "198.199.98.108:33445";
+    }
+    { pk = "A04F5FE1D006871588C8EC163676458C1EC75B20B4A147433D271E1E85DAF839";
+      addr = "52.53.185.100:33445";
+    }
+    { pk = "82EF82BA33445A1F91A7DB27189ECFC0C013E06E3DA71F588ED692BED625EC23";
+      addr = "tox.kurnevsky.net:33445";
+    }
+    { pk = "8EF12E275BA9CD7D56625D4950F2058B06D5905D0650A1FE76AF18DB986DF760";
+      addr = "tox.yikifish.com:33445";
+    }
+    { pk = "EDBEF0BE30CE1978F6EE9E13C1D4133409908929CC8AD7F56C4AE865B15B3177";
+      addr = "52.27.228.84:33445";
+    }
+  ];
+
+  configFile = pkgs.writeText "config.yml" (
+    builtins.toJSON {
+      log-type = cfg.logType;
+      keys-file = cfg.keysFile;
+      udp-address = cfg.udpAddress;
+      tcp-addresses = cfg.tcpAddresses;
+      tcp-connections-limit = cfg.tcpConnectionLimit;
+      lan-discovery = cfg.lanDiscovery;
+      threads = cfg.threads;
+      motd = cfg.motd;
+      bootstrap-nodes = bootstrapNodes;
+    }
+  );
+in {
+  options.services.tox-node = {
+    enable = mkEnableOption "Tox Node service";
+
+    logType = mkOption {
+      type = types.string;
+      default = "Stderr";
+    };
+    keysFile = mkOption {
+      type = types.string;
+      default = "/var/lib/tox-node/keys";
+    };
+    udpAddress = mkOption {
+      type = types.string;
+      default = "0.0.0.0:33445";
+    };
+    tcpAddresses = mkOption {
+      type = types.listOf types.string;
+      default = [ "0.0.0.0:33445" ];
+    };
+    tcpConnectionLimit = mkOption {
+      type = types.int;
+      default = 8192;
+    };
+    lanDiscovery = mkOption {
+      type = types.bool;
+      default = true;
+    };
+    threads = mkOption {
+      type = types.int;
+      default = 1;
+    };
+    motd = mkOption {
+      type = types.string;
+      default = "Hi from tox-rs! I'm up {{uptime}}. TCP: incoming {{tcp_packets_in}}, outgoing {{tcp_packets_out}}, UDP: incoming {{udp_packets_in}}, outgoing {{udp_packets_out}}";
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    users.users."tox-node" = {
+      description = "Tox node daemon user";
+      uid = config.ids.uids.tox-bootstrapd;
+    };
+
+    systemd.services.tox-node = {
+      description = "Tox Node";
+
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+
+      serviceConfig = {
+        ExecStart = "${pkg}/bin/tox-node config ${configFile}";
+        User = "tox-node";
+        Restart = "always";
+        PermissionsStartOnly = true;
+      };
+
+      preStart = ''
+        mkdir -m 700 -p ${homeDir}
```

Comment:
On my end, `mkdir` will not update the mask if the dir already exists. Consider using `install` or explicitly `chmod` it.